truffle/src/main/java/org/jruby/truffle/core/klass/ClassNodes.java

```diff
@@ -330,6 +330,8 @@ public SuperClassNode(RubyContext context, SourceSection sourceSection) {
         }
 
         @Specialization
+        @TruffleBoundary
+        // TODO (pitr-ch 22-Feb-2016): should it be on fast path?
```

Comment:
If it is, then we should have an additional field to keep the superclass in classes since it can only ever be changed by `Class#initialize` and would obviate the need for the lookup.